### ISSUE TITLE
Trailer not moving to combine when combine is 80% or more full

### ISSUE THREAD

**Issue by coatsy35** [Stephan-S/FS22_AutoDrive#929]
Seam to be having an issue with the trailer approaching a combine when it's nearly full it holds off approaching and then just sits there and the combine eventually stops with it's auger out. Sometime the trailer doesn't move again to the combine and I have to force it along, some time it does eventually move to the combine. Is this a known bug? is it likely to be fixed anytime soon?

**Comment by coatsy35** [comment]
#671 #640 #638 basically all still exist and cause issues. Do you need more info? debug etc... let me know and I'll get it as I'm having a nightmare on a large field on the Oaks (field 51) due to the excessive wait time at over 80% and then reversing and sometime not unloading at all and then when the trailer does get under the auger the combine moves forward but the trailer stays stationary when it shudl unload on the move etc...

**Comment by Iwan1803** [comment]
On large fields try to set the option to call an unloader to a lower value (60% is the default). Then the unloder will be called earlier and the problem shouldn't exist.
When CP drives the harvester, CP creates a parking pocket for itself (starting at 95%, if I'm not mistaken) to avoid unloading in/over the fruit. So that AD can then still reach the thresher without problems, AD stops.
I don't know if CP can communicate this behavior to AD.  Then maybe this 90% limit is not needed anymore.

**Comment by coatsy35** [comment]
> On large fields try to set the option to call an unloader to a lower value (60% is the default). Then the unloader will be called earlier and the problem shouldn't exist. When CP drives the harvester, CP creates a parking pocket for itself (starting at 95%, if I'm not mistaken) to avoid unloading in/over the fruit. So that AD can then still reach the thresher without problems, AD stops. I don't know if CP can communicate this behavior to AD. Then maybe this 90% limit is not needed anymore.

Ok. @tensuko @pvaiko would know more about that I'm sure. I often see the trailer stop before the combine even on a long row at times where no pocket is required. The trailer will be travelling towards the combine and then just slows down and stops a long distance away from the combine for no reason as the combine reaches full. The combine will then just sit there with the unloading auger out and the trailer won't move. (I always use AD for combine unloading in combine mode). Other times the trailer gets called at the precall trigger and makes it's way across the field or around the headland slowing down and pausing as it goes and then as the combine fills it also just stops and waits doing nothing. Maybe there is a conflict somewhere?

Also setting the precall really low would have the combine with it's unloading auger out for a lot of the time, hence my enhancement ide of not having the unloading auger open until the trailer is near to the rear of the combine and nearly there

**Comment by pvaiko** [comment]
> I don't know if CP can communicate this behavior to AD.

CP provides API functions for AD to tell if the combine is waiting for unload in a pocket or after it pulled back (because the pipe was in the fruit), or just simply waiting for unload. We also tell AD if we are maneuvering (like creating the pocket or pulling back) so it can stay away until we are done.

**Comment by coatsy35** [comment]
Classic example here, combine running down a nice straight run, coming up to 95% full, trailer not far away but is not chasing it, just sat still even though precall has been raised. Combine stops to unload as trailer not under pipe, trailer stops before it gets to combine and then starts reversing away! stops and then thinks about it, then eventually moves away and positions itself under the pipe, combine starts to move but the trailer doesn't so combine stops again and then has to unload stationary when it should me unloading on the move.

![fsScreen_2023_07_07_20_47_40](https://github.com/Stephan-S/FS22_AutoDrive/assets/122838698/ccaa9e9d-9c33-48e3-8989-dd6e8cca2145)

**Comment by coatsy35** [comment]
[log.txt](https://github.com/Stephan-S/FS22_AutoDrive/files/11985704/log.txt)

![fsScreen_2023_07_07_21_16_24](https://github.com/Stephan-S/FS22_AutoDrive/assets/122838698/368f4c0a-1085-4c69-8f34-d9ca791a6807)

**Comment by coatsy35** [comment]
Little video of it not starting to unload on the move, goes under pipe starts t move then stops and has to unload stationary 

https://youtu.be/MOcgRt5l6f0

**Comment by coatsy35** [comment]
Another one showing some weird unloading action, just not catching up very quick, like it's hesitating or stuck.

https://youtu.be/Fh51K7HOPE8

**Comment by coatsy35** [comment]
Driving through crop and crazy manoeuvres

https://youtu.be/HEPrRIFyqG4

**Comment by coatsy35** [comment]
Not getting under the pipe quick enough whilst the combine is moving, then reversing up and pausing, then moving under the pipe but not moving forcing a stationary unload

https://youtu.be/7gi7VbWOql4

**Comment by coatsy35** [comment]
More nonsense out in the middle of a large field. There is obviously an issue here, but what it is I cannot say, it's like it gets stuck in a loop of going wrong, left alone enough it will fix itself. However the reversing up ones perplex me.

https://youtu.be/nwc1WPbbd3o

and

https://youtu.be/U1yImGPDgpI

**Comment by rohne83** [comment]
Only a guess: in the past, if combine stopped in the middle of the lane because full, AD somehow prevented it from moving again until unloading completely finished (was often criticized). In your video this suppression doesn't happen but combine starts moving again (in fact that's what most of us want, but doesn't seem to work correctly atm). Maybe the cancelation of the unloading is related to this former behaviour.

For the 80% threshold, there are two different ones: 80% is the fill level above which now new unloader will be called. Additionally there's a second threshold for cases where unloader was called (with combine fill level still below 80% ofc) but doesn't manage to get to the combine quickly - if combine reaches 90% the unloader will stop chasing the combine. You see the behaviour of this 90% threshold during the first 10 seconds of your first video of your last comment.

**Comment by coatsy35** [comment]
> Additionally there's a second threshold for cases where unloader was called already (with combine fill level still below 80% ofc) but doesn't manage to get to the combine quickly - if combine reaches 90% in the meantime, the unloader will stop chasing the combine. You see the behaviour of this 90% threshold during the first 10 seconds of your fir

@rohne83 yes I agreed. That's why I also wanted an option to keep the unloader in on the precall. The combine doesn't' need to put the unloader out until the trailer is either next to the combine or nearly approaching it from the rear - not from the time the precall is actioned. As CP has an API to notify now what it is doing with pockets etc... I would think this behaviour can also be fixed. If the trailer has started chasing the combine, then it shouldn't stop at all even if the combine is atr 100% and then when it gets under the pipe it should start unloading on the move as the combine starts moving forward again. I hope they fix this in the next update.
